[[ Bug 21245 ]] SE indent errors with inline block comments

bwmilby opened the pull request:
There are 2 related errors in the `lineStripComments` /` __StripComment` handlers in the `revsecommoneditorbehavior.livecodescript` code.

The first error is that "/*" is assumed to run to the end of the current line.  Since it is possible to have multiple inline block comments on a single line, a recursive call to `__StripBlockComment` is used to remove these comments.

The second error is that the loop that goes through the current line does not exit when a true comment to the end of the line is encountered.  The loop uses quotes to ensure that it isn't capturing comment characters in a string, but fails to exit.  Added an additional return value (pass by reference variable) indicating comment status of the current chunk to better facilitate knowing when to exit the loop.

bwmilby committed:
[[ Bug 21245 ]] SE indent errors with inline block comments

There are 2 related errors in the lineStripComments / __StripComment handlers in the revsecommoneditorbehavior.livecodescript code.

The first error is that "/*" is assumed to run to the end of the current line.  Since it is possible to have multiple inline block comments on a single line, a recursive call to __StripBlockComment is used to remove these comments.

The second error is that the loop that goes through the current line does not exit when a true comment to the end of the line is encountered.  The loop uses quotes to ensure that it isn't capturing comment characters in a string, but fails to exit.  Added an additional return value (pass by reference variable) indicating comment status of the current chunk.

montegoulding commented:
@bwmilby could you add some indentation tests for the fix to `tests/scripteditor/_indentation_tests`

bwmilby committed:
[[ Bug 21245 ]] Add indent tests

Add 2 test cases for issues fixed by the patch.
- inline block comments before line continuation
- quote characters in a comment could cause a `\` to be parsed as a line continuation character

PASS:    /Users/milby/projects/GitHub/livecode/ide/tests/scripteditor/indentation: TestIndents [9/9]

montegoulding commented:
@livecode-vulcan review ok 6373c36

livecode-vulcan commented:
💙 review by @montegoulding ok 6373c36f8d69c48737d04a1c5a036867f25086da
